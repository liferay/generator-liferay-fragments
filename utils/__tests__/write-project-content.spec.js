const getProjectContent = require('../get-project-content');
const getTestFixtures = require('../get-test-fixtures');
const tmp = require('tmp');
const writeProjectContent = require('../write-project-content');

describe('utils/write-project-content', () => {
  getTestFixtures().map(projectPath =>
    it('writes a project inside a given path', async () => {
      const tmpDir = tmp.dirSync({ unsafeCleanup: true });
      const projectContent = getProjectContent(projectPath);

      await writeProjectContent(tmpDir.name, projectContent);

      const newProjectContent = getProjectContent(tmpDir.name);
      tmpDir.removeCallback();

      expect(projectContent.collections).toEqual(newProjectContent.collections);
    })
  );
});
