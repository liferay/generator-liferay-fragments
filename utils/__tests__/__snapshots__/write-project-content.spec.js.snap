// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/write-project-content needs a valid projectBasePath parameter 1`] = `[AssertionError [ERR_ASSERTION]: projectBasePath must be a valid filesystem path]`;

exports[`utils/write-project-content needs a valid projectContent parameter 1`] = `[AssertionError [ERR_ASSERTION]: projectContent must be a valid project object]`;
